sklearn/datasets/tests/test_base.py

```diff
@@ -347,3 +349,16 @@ def test_load_boston_alternative():
 
     np.testing.assert_allclose(data, boston_sklearn.data)
     np.testing.assert_allclose(target, boston_sklearn.target)
+
+
+def test_fetch_remote_raise_warnings_with_invalid_url():
+    from urllib.error import HTTPError
+
+    invalid_remote_file = RemoteFileMetadata(
+        "invalid_file", "https://scikit-learn.org/this_file_does_not_exist.tar.gz", None
+    )
+
+    with pytest.warns(UserWarning, match="Retry downloading") as record:
+        with pytest.raises(HTTPError, match="HTTP Error 404"):
+            _fetch_remote(invalid_remote_file, n_retries=3, delay=0)
+        assert len(record) == 3
```

Comment:
Can you please check that the content of the warning messages is the one we expect with something like:

```python
expected_message = (
    "UserWarning: blah blah blah"
    "more blah blah blah"
)
for warning in record:
     assert warning.message == expected_message
```

I am not 100% sure of the pytest warning record API, so this might require some adjustment but you get the idea.

Reply:
Hi @ogrisel , I've added the tests as you suggested :)